record/record.app.js

```diff
@@ -50,6 +52,8 @@
                     $rootScope.relatedReferences = $rootScope.reference.related;
                     // There should only ever be one entity related to this reference
                     return $rootScope.reference.read(1);
+                }, function error(exception) {
+                    return $q.reject(exception);
```

Comment:
I hadn't noticed I missed this. It looks like if that first promise errored out it would do so gracefully without this. With `$q.reject()`, did you test that it calls the `.then()` on the next line and then goes straight to the `error` callback where it throws the exception?